rotation_first: true
rotation_while_explore: true